types/network.go

```diff
@@ -61,6 +60,31 @@ type ConnectionConfig struct {
 	InsecureSkipVerify bool
 	// UseRoundRobin
 	UseRoundRobin bool
+	// ParralelismConfig determines how many concurrent connections to have.
+	Parralelism ParralelismConfig
+}
+type ParralelismConfig struct {
+	// AllowedDriftSeconds is the maximum amount that is allowed before the connections scales up.
+	AllowedDriftSeconds int64
+	// MinimumScaleDownDriftSeconds is the amount if we go below that we can scale down.
+	MinimumScaleDownDriftSeconds int64
+	// MaxConnections is the maximum number of concurrent connections to use.
+	MaxConnections uint
+	// MinConnections is the minimum number of concurrent connections to use.
+	MinConnections uint
```

Comment:
We probably want a check to make sure it's not 0? currently I think in face of errors we decrese shards. What if it goes to 0?

Reply:
I avoided adding validation since the PR was already huge. I will add it either in alloy or here in a separate pr.